test/test_filesystem_helper.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "rcpputils/filesystem_helper.hpp"
+
+using path = rcpputils::fs::path;
+
+TEST(TestFilesystemHelper, join_path)
+{
+  auto p = path("foo") / path("bar");
+
+#ifdef _WIN32
```

Comment:
To minimize the risk of having Windows-only compilation errors, I'd suggest doing the following:

```c++
#ifdef WIN32
static constexpr const bool is_win32 = true;
#else
static constexpr const bool is_win32 = false;
#endif

// [...]

TEST(...) {
  // => now both sides of what was an #ifdef are built on Linux so no code paths are left untested.
  if (is_win32) {
    // ...
  } else {
    // ...
  }
}
```

This is a very minor issue, but that'd be great to avoid ending up with ifdef's everywhere once we start writing more tests.

Reply:
Good call!